### ISSUE TITLE
Feasibility of implementing a Rust software HAL target? Any WIP efforts out there?

### ISSUE THREAD

**Issue by mitchmindtree** [gfx-rs/gfx#3113]
I was just wondering what the feasibility of implementing a software-renderer target for gfx-hal would be, and whether or not there are any WIPs out there? This would be amazingly useful for testing/CI/debugging and perhaps as a reference implementation, but I can imagine it might be a massive amount of work?

**Comment by kvark** [comment]
I agree, that would be very useful! Also, that would be insanely difficult to get right.
Related efforts I'm aware of:
  - https://swiftshader.googlesource.com/SwiftShader/ - official Vulkan in software by Google
  - https://github.com/jrmuizel/sw-wr - software WebRender backend in development

As you can see, neither is exactly the subject, so there is still a room for a solution here.
